### PR TITLE
Add etcd sequencer

### PR DESCRIPTION
**Pull request by stlpmo-jn**
store the fid sequence in ETCD,  the master require a range of sequences from ETCD for dir/assign operation.

resolved problem:
when the master leader changed,   it will take a while for the new leader to recieve the max fid sequence from volume servers. In this duration, the operation dir/assign will return duplicate fid.

### CODE REVIEW

**Comment by stlpmo-jn** [comment]
i have no idea about the check failed. 
need help

File: weed/command/master.go [hunk 1]
```diff
@@ -55,6 +58,9 @@ func init() {
 	m.disableHttp = cmdMaster.Flag.Bool("disableHttp", false, "disable http requests, only gRPC operations are allowed.")
 	m.metricsAddress = cmdMaster.Flag.String("metrics.address", "", "Prometheus gateway address")
 	m.metricsIntervalSec = cmdMaster.Flag.Int("metrics.intervalSeconds", 15, "Prometheus push interval in seconds")
+	m.sequencerType = cmdMaster.Flag.String("sequencerType", "memory", "Choose [memory|etcd] type for store the file sequence")
```
Comment (chrislusf) [inline weed/command/master.go]: moving this config to `weed scaffold -config=master` will be easier for later other kind of sequence type and configurations.

Comment (ghost) [reply weed/command/master.go]: ok, good idea, thank you

**Comment by stlpmo-jn** [comment]
move the parameters from cmd-line to the file master.toml

File: weed/util/config.go [hunk 2]
```diff
@@ -40,5 +42,19 @@ func LoadConfiguration(configFileName string, required bool) (loaded bool) {
 	}
 
 	return true
+}
 
+func Config() Configuration {
+	return viper.GetViper()
 }
+
+func SubConfig(subKey string) (Configuration, error) {
```
Comment (chrislusf) [inline weed/util/config.go]: not used?

File: weed/util/config.go [hunk 3]
```diff
@@ -40,5 +42,19 @@ func LoadConfiguration(configFileName string, required bool) (loaded bool) {
 	}
 
 	return true
+}
 
+func Config() Configuration {
```
Comment (chrislusf) [inline weed/util/config.go:45]: may not need this function since it is called just twice.

File: weed/util/config.go (hunk 2 above)
Comment (ghost) [reply weed/util/config.go]: this function has not be used.

File: weed/util/config.go (hunk 3 above)
Comment (ghost) [reply weed/util/config.go:45]: the interface Configuration should be the facade of any configuration (including viper),  it should isolate the viper from out project ( seaweeddfs), we should  encapsulate it completly.
this is my opinion.

File: weed/command/scaffold.go [hunk 4]
```diff
@@ -346,5 +346,12 @@ scripts = """
 """
 sleep_minutes = 17          # sleep minutes between each script execution
 
+sequencer_type = memory     # Choose [memory|etcd] type for storing the file id sequence
```
Comment (chrislusf) [inline weed/command/scaffold.go:349]: should be double quoted as "memory"

File: weed/command/scaffold.go [hunk 5]
```diff
@@ -346,5 +346,12 @@ scripts = """
...
+
+# when sequencer.type = etcd, set listen client urls of etcd cluster that store file id sequence
+# example : http://127.0.0.1:2379,http://127.0.0.1:2389
+sequencer_etcd_urls = http://127.0.0.1:2379
```
Comment (chrislusf) [inline weed/command/scaffold.go:353]: should be double quoted also

**Pull request merged**